### PR TITLE
chore: Add pre-commit hook checks and configuration

### PR DESCRIPTION
**Pull request by Tiboris**
Hello sorry to make noise with first commit i just added several pre-commit hook check for yamls and python files.
I enabled black, yamlint and some more features can be enabled if desired.

Main reason to create this PR is to prove that we can use container image with PR number tag (later built in first pipeline step and pushed with PR number tag)

### CODE REVIEW

**Comment by Tiboris** [comment]
This action update and checks inside are POC that action can be setup with custom image based on PR number:
https://github.com/freeipa/ipa-tuura/actions/runs/3695297690/jobs/6257520267

see: 
```yaml 
test:
    name: test the built image
    runs-on: "quay.io/neoave/bridge:${{github.event.number}}"
```
https://github.com/freeipa/ipa-tuura/pull/15/files#diff-62a47943e508f5e7600aef291d8dee4a5cd2a17f806d69a032721e4087277b18R30

**Comment by Tiboris** [comment]
FYI 3rd commit is just a scratch it can be removed and your pipeline will be enhanced a bit with checks created in other 2 commits. This is an enhancement to get your code tested more in MR and even before that with enabling pre-commit hooks on your local forks clones.

**Comment by Tiboris** [comment]
secrets for quay account bot must be added for this PR to work
see working version: https://github.com/Tiboris/ipa-tuura/pull/3 with proper quay credentials

File: src/ipa-tuura/ipatuura/migrations/0001_initial.py [hunk 1]
```diff
@@ -1,51 +1,150 @@
 # Generated by Django 4.0.2 on 2022-10-14 16:10
 
-from django.db import migrations, models
```
Comment (f-trivino) [inline src/ipa-tuura/ipatuura/migrations/0001_initial.py:3]: the migrations folder is generated by the django framework when running makemigrations command, I think we should skip format of the generated files.

**Pull request closed without merging**